MarkBind/markbind
# Merge /vue-strap into /markbind

**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [x] Other, please explain:

Resolves #411 
Resolves #981 

**What is the rationale for this request?**
Merge /vue-strap repo into /markbind

**What changes did you make? (Give an overview)**
I aimed for minimal changes for now, but one thing lead to another as explained in the commit messages (particularly the last one), and had to make quite a few changes still.

**Is there anything you'd like reviewers to focus on?**
na

**Testing instructions:**
- `npm run build` should build the minified `vue-strap` as per normal
-  `npm run test/lint` should pass, and the snapshot tests should also be run

**Proposed commit message: (wrap lines at 72 characters)**
Merge /vue-strap into /markbind

Markbind's vue component definitions are stored in the vue-strap fork.
This has some implications on development and maintainer workflow (see
issue #411 for more detail), most notably needing to manually copy the
minified bundle to test changes in /vue-strap.

An alternative was raised in issue #981 to alleviate this, that is to
make /vue-strap a scoped package. This still has several drawbacks as
compared to a complete repo merge as outlined in #411, such as needing
to run tests on both repos.

Hence, let's change the directory structure and configurations of
/markbind to accomodate the linting, testing and building processes
from /vue-strap.
